lib/validator.js

```diff
@@ -155,9 +171,104 @@ const getValidator = (rules) => {
   }
 };
 
+
+
+/**recast to a string
+ * 
+ * @param input
+ * @returns {string}
+ */
+const stringCaster = (input) => {
+  const type = typeof(input);
+  if(type === 'string') {
+    return input;
+  } else if(type === 'number' || type === 'boolean') {
+    const coercedInput = String(input);
+    return coercedInput;
+  } else if(type === 'object' && String(input).match(/Time\)$/)) {
+    const coercedInput = String(input);
+    return coercedInput;
+  }
+  throw new CastError('string', input);
+};
+
+/**recast to a number
+ * 
+ * @param input
+ * @returns {number}
+ */
+const numberCaster = (input) => {
+  const type = typeof(input);
+  if(type === 'number') {
+    return input;
+  } else if(input === '') {
+    throw new CastError('number', input);
+  } else if(type === 'string' || type === 'date') {
+    const coercedInput = Number(input);
+    if(!isNaN(coercedInput)) {
+      return coercedInput;
+    } else {
+      throw new CastError('number', input);
+    }
+  }
+  throw new CastError('number', input);
+};
+
+/**recast to a boolean
+ * 
+ * @param input
+ * @returns {boolean}
+ */
+const booleanCaster = (input) => {
+  const type = typeof(input);
+  if(type === 'boolean') {
+    return input;
+  } else if(input === 'true' || input === 1 || input === '1') {
+    return true;
+  } else if(input === 'false' || input === 0 || input === '0') {
+    return false;
+  } 
+  throw new CastError('boolean', input);
+};
+
+/**recast to a date
+ * 
+ * @param input
+ * @returns {date}
+ */
+const dateCaster = (input) => {
+  const type = typeof(input);
+  if(type === 'string' && input.match(/Time\)$/)) {
+    return input;
+  }
+  throw new CastError('date', input);
+};
+
+/**
+ * Based on what type you provide, return the appropriate caster
+ * @param type
+ * @returns {caster}
+ */
+
+const getCaster = (type) => {
+  if(type === 'string'){
+    return stringCaster;
+  }
+  if(type === 'number'){
+    return numberCaster;
```

Comment:
sort of like with the last lab, this logic totally works, but I feel like an object or switch would be a little cleaner here!